CONTRIBUTING.md

```diff
@@ -29,3 +29,92 @@ Recommendations and requirements for how to best contribute to Parquet. We striv
+Due to confusion in the past over parquet versioning it is not expected that there will be a 3.0 release of the specification in the foreseeable future.
+
+### Compatibility and Feature Enablement
+
+For the purposes of this discussion we classify features into the following buckets:
+
+1. Backwards compatible.  A file written by an older version of a library can be read by a newer version of the
+library.
+
+2. Forwards compatible.  A file written by a new version of the library can be read by an older version
+of the library. 
+
+3. Forward compatible with suboptimal performance. A file written by a new version of the library can
+be read an older version of the library but performance might be suboptimal (e.g. statistics might be missing
+from the older reader's perspective).
+
+4. Forward incompatible. A file written with a new version of the library cannot be read by an older version
+of the library.
+
+The Parquet community hopes that new features are widely beneficial
+to users of Parquet, and therefore third-party implementations will
+adopt them quickly after they are introduced. It is assumed that most new features will be implemented behind a feature flag that defaults to "off".To avoid, compatibility issues across the ecosystem some amount of lead time is desirable to ensure a critical mass of Parquet implementations support a feature.  Therefore, the Parquet PMC gives the following guidance for changing a feature to be "on" by default:
+
+1. Backwards compatibility is the concern of implementations but given the ubiquity of Parquet and the length
+of time it has been used, libraries SHOULD support reading older
+file variants.
+
+2. Forward compatible changes MAY be used by default in implementations once the parquet-format containing
+those changes has been formally released. These features SHOULD be turned on 1 year after the parquet-java
+implementation containing feature is released (e.g. it is expected
+the Java implementation itself will turn them on for the first
+release 1 year after a features initial introduction).
+
+3. Forward compatible with suboptimal performance features MAY be made default after 
+the parquet-java implementation containing the feature is released. Features in this category
+SHOULD be turned on 1 year after the parquet-java
+implementation containing the feature is released.  Implementations MAY choose
+to do a major version bump when turning on a feature in this category.
+
+4. Forwards incompatible changes MAY be made default 2 years after the parquet-java
+implementation containing the feature is released. Features in this category SHOULD be turned on by 
+default 3 years after the parquet-java implementation containing feature is released. Implementations MUST do 
+a major version bump when enabling a forward incompatible feature by default.
```

Comment:
I think this is general guidance and each can do there own versioning.  For parquet-java I think this will hopefully be binding.  As it stands I believe parquet-rs and parquet-cpp are effectively doing major version releases on each release.

Reply:
this language has been walked back further, I think it tries to make clear that versioning is not a concern of the recommendations.